stripe_test.go

```diff
@@ -165,6 +166,207 @@ func TestDo_RetryOnTimeout(t *testing.T) {
 	assert.Equal(t, uint32(2), atomic.LoadUint32(&counter))
 }
 
+// Test that telemetry metrics are not sent by default
+func TestDo_TelemetryDisabled(t *testing.T) {
+	type testServerResponse struct {
+		Message string `json:"message"`
+	}
+
+	message := "Hello, client."
+	requestNum := 0
+
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// none of the requests should include telemetry metrics
+		assert.Equal(t, r.Header.Get("X-Stripe-Client-Telemetry"), "")
+
+		response := testServerResponse{Message: message}
+
+		data, err := json.Marshal(response)
+		assert.NoError(t, err)
+
+		_, err = w.Write(data)
+		assert.NoError(t, err)
+
+		requestNum++
+	}))
+	defer testServer.Close()
+
+	backend := stripe.GetBackendWithConfig(
+		stripe.APIBackend,
+		&stripe.BackendConfig{
+			LogLevel:          3,
+			MaxNetworkRetries: 0,
+			URL:               testServer.URL,
+		},
+	).(*stripe.BackendImplementation)
+
+	for i := 0; i < 2; i++ {
```

Comment:
It's initially unclear why this loop executes twice.  Can you add a quick comment that explains that it's because the telemetry comes from the _previous_ completed request/response?